contribs/gmf/src/drawing/drawFeatureOptionsComponent.js

```diff
@@ -0,0 +1,517 @@
+
+    /**
+     * @type {?import("ngeo/interaction/MeasureLength.js").default}
+     */
+    this.measureLength = null;
+
+    /**
+     * @type {?import("ngeo/interaction/MeasureArea.js").default}
+     */
+    this.measureArea = null;
+
+    /**
+     * @type {?import("ngeo/interaction/MeasureAzimut.js").default}
+     */
+    this.measureAzimut = null;
+
+    /**
+     * @type {?OLInteractionDraw}
+     */
+    this.drawRectangle = null;
+
+    // === Injected properties ===
+
+    /**
+     * @type {angular.IScope}
+     * @private
+     */
+    this.scope_ = $scope;
+
+    // === Inner properties ===
+
+    /**
+     * The length is used to determine:
+     * - the length of a line segment to draw (for length and area measurements)
+     * - the width of a rectangle to draw
+     * - the radius of a circle to draw (for azimut measurements)
+     * @type {number}
+     */
+    this.length = NaN;
+
+    /**
+     * Measure unit for the length.  Possible values are: 'm', 'km'.
+     * @type {string}
+     */
+    this.lengthUnits = 'm';
+
+    /**
+     * The height is used to determine:
+     * - the height of the rectangle to draw
+     * @type {number}
+     */
+    this.height = NaN;
+
+    /**
+     * Measure unit for the height.  Possible values are: 'm', 'km'.
+     * @type {string}
+     */
+    this.heightUnits = 'm';
+
+    // === Private properties ===
+
+    /**
+     * The draw interaction that was given.
+     * @type {?OLInteractionDraw}
+     * @private
+     */
+    this.drawInteraction_ = null;
+
+    /**
+     * The feature being drawn. Set when the 'drawstart' event is
+     * fired by the draw interaction.
+     * @type {?OLFeature<import("ol/geom/Geometry.js").default>}
+     * @private
+     */
+    this.feature_ = null;
+
+    /**
+     * @type {import("ol/events.js").EventsKey[]}
+     * @private
+     */
+    this.listenerKeys_ = [];
+
+    /**
+     * Used to keep track of the current number of vertices while
+     * drawing a LineString or a Polygon. Why? Because the only way to
+     * keep track of the changes that occur within their geometry is
+     * with the 'change' event, which occurs while the mouse is
+     * moving.
+     *
+     * If at any moment, while drawing, this number becomes different
+     * than the actual number of vertices in the geometry, then we
+     * need to update the snap feature.
+     * @type {number}
+     */
+    this.verticesCounter_ = 2;
+
+    /**
+     * @type {!OLFeature<import("ol/geom/Geometry.js").default>}
+     * @private
+     */
+    this.snapFeature_ = new OLFeature();
+
+    /**
+     * @type {!OLSourceVector<import("ol/geom/Geometry.js").default>}
+     * @private
+     */
+    this.snapSource_ = new OLSourceVector({
+      features: [this.snapFeature_]
+    });
+
+    /**
+     * @type {!OLInteractionSnap}
+     * @private
+     */
+    this.snapInteraction_ = new OLInteractionSnap({
+      // @ts-ignore: webpack
```

Comment:
I had to add this line to ignore an error reported by the typecheck.  I couldn't find why or what it was complaining about.  Any hint would be welcome.

Reply:
the `handleEvent` option is missing in OpenLayers, that's fine to have a `ts-ignore`